src/hooks/useAllPairsWithNonExpiredLiquidityMiningCampaignsAndLiquidityAndStakingIndicator.ts

```diff
@@ -1,18 +1,21 @@
-import { gql, useQuery } from '@apollo/client'
+import { gql } from 'graphql-request'
 import Decimal from 'decimal.js-light'
 import { CurrencyAmount, Pair, Token, TokenAmount, USD } from '@swapr/sdk'
 import { getAddress, parseUnits } from 'ethers/lib/utils'
 import { DateTime, Duration } from 'luxon'
-import { useMemo } from 'react'
+import { useEffect, useMemo, useState } from 'react'
 import { useActiveWeb3React } from '.'
 import { SubgraphLiquidityMiningCampaign } from '../apollo'
 import { useAllTokensFromActiveListsOnCurrentChain } from '../state/lists/hooks'
 import { toLiquidityMiningCampaigns } from '../utils/liquidityMining'
 import { useNativeCurrency } from './useNativeCurrency'
+import { immediateSubgraphClients } from '../apollo/client'
+
+const PAGE_SIZE = 1000
 
 const QUERY = gql`
```

Comment:
Just a thought for a future version. We could put all queries into one place. In Aqua, they're all in `src/subgraph/queries.ts`.